modules/frontend/queue/queue.go

```diff
@@ -197,13 +190,22 @@ FindQueue:
 	goto FindQueue
 }
 
-func (q *RequestQueue) forgetDisconnectedQueriers(_ context.Context) error {
+func (q *RequestQueue) cleanupQueues(_ context.Context) error {
 	q.mtx.Lock()
 	defer q.mtx.Unlock()
 
-	if q.queues.forgetDisconnectedQueriers(time.Now()) > 0 {
-		// We need to notify goroutines cause having removed some queriers
-		// may have caused a resharding.
+	removedQueue := false
+
+	// look for 0 len queues and remove them
+	for userID, uq := range q.queues.userQueues {
+		if uq.ch != nil && len(uq.ch) == 0 {
```

Comment:
I think  we can simplify this:
```go
if  len(uq.ch) == 0
```
It's not possible to be nil because we always create the channel with make. Furthermore, len also handle nil.